src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java

```diff
@@ -0,0 +1,321 @@
+                return FormValidation.ok();
+            try {
+                final String matlabRoot = Jenkins.getInstance()
+                        .getDescriptorByType(AddMatlabToPathDescriptor.class).getMatlabRootFolder();
+                for (Function<String, FormValidation> val : validations) {
+                    FormValidation validationResult = val.apply(matlabRoot);
+                    if (validationResult.kind.compareTo(Kind.ERROR) == 0
+                            || validationResult.kind.compareTo(Kind.WARNING) == 0) {
+                        return validationResult;
+                    }
+                }
+            }catch (Exception e) {
+                return FormValidation.warning(Message.getValue("Builder.invalid.matlab.root.warning"));
+            }
+           
+            return FormValidation.ok();
+        }
+    }
+
+    @Override
+    public void perform(@Nonnull Run<?, ?> build, @Nonnull FilePath workspace,
+            @Nonnull Launcher launcher, @Nonnull TaskListener listener)
+            throws InterruptedException, IOException {
+        //Set the environment variable specific to the this build
+        setEnv(build.getEnvironment(listener));
+        
+        // Invoke MATLAB command and transfer output to standard
+        // Output Console
+
+        buildResult = execMatlabCommand(workspace, launcher, listener, getEnv());
+
+        if (buildResult != 0) {
+            build.setResult(Result.FAILURE);
+        }
+    }
+
+    private synchronized int execMatlabCommand(FilePath workspace, Launcher launcher,
+            TaskListener listener, EnvVars envVars) throws IOException, InterruptedException {
+        ProcStarter matlabLauncher;
+        try {
+            matlabLauncher = getProcessToRunMatlabCommand(workspace, launcher, listener, envVars,
+                    constructCommandForTest(getInputArguments()));
+
+            // Copy MATLAB scratch file into the workspace.
+            FilePath targetWorkspace = new FilePath(launcher.getChannel(), workspace.getRemote());
+            copyFileInWorkspace(MatlabBuilderConstants.MATLAB_RUNNER_RESOURCE,
+                    MatlabBuilderConstants.MATLAB_RUNNER_TARGET_FILE, targetWorkspace);
+        } catch (Exception e) {
+            listener.getLogger().println(e.getMessage());
+            return 1;
+        }
+        return matlabLauncher.join();
+    }
+    
+    public String constructCommandForTest(String inputArguments) {
+        String runCommand;
+        String matlabFunctionName = FilenameUtils.removeExtension(
+                Message.getValue(MatlabBuilderConstants.MATLAB_RUNNER_TARGET_FILE));
+        runCommand = "exit(" + matlabFunctionName + "(" + inputArguments + "))";
+        return runCommand;
+    }   
+    
+    // Concatenate the input arguments
+    private String getInputArguments() {
+        String pdfReport = MatlabBuilderConstants.PDF_REPORT + "," + this.getPdfReportChkBx();

```

Comment:
here and elsewhere, should "final" be used wherever possible? In C++ we always use 'const' unless we don't want a const.

Reply:
Done for all the identified methods